conn2res/reservoir.py

```diff
@@ -9,8 +9,223 @@
 import numpy as np
 import numpy.ma as ma
 from numpy.linalg import (pinv, matrix_rank)
+from scipy.linalg import eigh
+from bct.algorithms.clustering import get_components
+from bct.algorithms.distance import distance_bin
+
+from .iodata import load_file
+from .coding import get_modules
+
+
+class Conn:
+    """
+    Class that represents a connectivity matrix representing either weighted
+    or unweighted connectivity data
+
+    Attributes
+    ----------
+    # TODO
+
+    Methods
+    ----------
+    # TODO
+    """
+
+    def __init__(self, subj_id=0):
+        # load connectivity data
+        self.w = load_file('connectivity.npy')
+
+        # select one subject
+        self.w = self.w[:, :, subj_id]
+
+        # number of all active nodes
+        self.n_nodes = len(self.w)
+
+        # indexes of set of active nodes
+        self.idx_node = np.full(self.n_nodes, True)
+
+    def scale_and_normalize(self):
+        """
+        Scales the connectivity matrix between [0, 1] and divides by spectral
+        radius
+
+        # TODO
+        """
+
+        # scale connectivity matrix between [0, 1]
+        self.scale()
+
+        # divide connectivity matrix by spectral radius
+        self.normalize()
+
+    def scale(self):
+        """
+        Scales the connectivity matrix between [0, 1]
+
+        # TODO
+        """
+
+        # scale connectivity matrix between [0, 1]
+        self.w = (self.w - self.w.min()) / (self.w.max() - self.w.min())
+
+    def normalize(self):
+        """
+        Normalizes the connectivity matrix with spectral radius
+
+        # TODO
+        """
+
+        # divide connectivity matrix by spectral radius
+        ew, _ = eigh(self.w)
+        self.w = self.w / np.max(ew)
+
+    def binarize(self):
+        """
+        Binarizes the connectivity matrix
+
+        # TODO
+        """
+
+        # binarize connectivity matrix
+        self.w = self.w.astype(bool).astype(int)
+
+    def subset_nodes(self, node_set, **kwargs):
+        """
+        Defines subset of nodes of the connectivity matrix and reduces
+        the connectivity matrix to this subset
+
+        # TODO
+        """
+
+        # get nodes
+        idx_node = np.isin(np.arange(self.n_nodes),
+                           self.get_nodes(node_set, **kwargs))
+
+        # update node attributes
+        self._update_nodes(idx_node)
+
+        # update component
+        self._update_component()
+
+    def get_nodes(self, node_set, nodes_from=None, nodes_without=None, n_nodes=1, **kwargs):
```

Comment:
This method can be used to define/return the input_nodes, output nodes etc. of the connectivity matrix (later reservoir).

nodes_from defines a subset of nodes we want to select from
nodes_without defines a subset of nodes we want to exclude from the selection